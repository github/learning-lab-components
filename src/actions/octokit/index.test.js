const octokit = require('.')
const mockContext = require('../../../tests/mockContext')

describe('octokit', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      issues: { create: jest.fn() }
    })
  })

  it('calls the correct octokit method', async () => {
    await octokit(context, { method: 'issues.create', title: 'test', body: 'test', owner: 'JasonEtco', repo: 'example' })
    expect(context.github.issues.create).toHaveBeenCalled()
    expect(context.github.issues.create.mock.calls).toMatchSnapshot()
  })

  it('throws if the provided method does not exist', async () => {
    try {
      await octokit(context, { method: 'issues.nope' })
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })

  it('throws if the provided method is called without all the required parameters', async () => {
    try {
      // Omitting the `title` field
      await octokit(context, { method: 'issues.create', body: 'test' })
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
})
