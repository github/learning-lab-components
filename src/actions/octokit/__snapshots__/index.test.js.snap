// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`octokit calls the correct octokit method 1`] = `
Array [
  Array [
    Object {
      "body": "test",
      "owner": "JasonEtco",
      "repo": "example",
      "title": "test",
    },
  ],
]
`;

exports[`octokit throws if the provided method does not exist 1`] = `[Error: Octokit method \`issues.nope\` does not exist.]`;

exports[`octokit throws if the provided method is called without all the required parameters 1`] = `[Error: Octokit method \`issues.create\` requires the following parameters: owner, repo, title]`;
